app/lib/linters/crunch_api.rb

```diff
@@ -0,0 +1,97 @@
+require 'rest-client'
+require 'base64'
+module Linters
+
+  # A class called Crunch
+  class CrunchApi
+
+    attr_accessor :base_url
+
+    def initialize()
+      @base_url = "https://platform.42crunch.com"
+    end
+
+    # Lints an unploaded file with 42crunch.
+    #
+    # @param file [File] the file to be linted
+    # @return [String] the raw JSON response
+    def lint_to_json(file)
+
+      # Initialize local variables
+      create_response = String.new
+      created_api_id = String.new
+      report_response = String.new
+      report_data = String.new
+      decoded_report = String.new
+
+      # Post the file to 42Crunch according to their API v1 "Create an API (from file)" spec
+      # https://www.postman.com/get-42crunch/workspace/42crunch-api/request/13761657-335bcb36-c68f-43a1-823d-741b62b55bc6
+      begin
```

Comment:
The presence of these `begin` blocks is a strong indication that this method is doing too much. 

A much better way to code this would be to break the sections `create_response`, `created_api_id`, `report_response`, `report_data`, `decoded_report` into separate methods.

The `lint_to_json` method could then call each of the methods in turn as required.

It would then also be much easier to test each of these actions individually.